### PR TITLE
✅  New decimal-aware price system - Part 9: Fix Misc & Async tests

### PR DESCRIPTION
**Pull request by JuaniRios**
Main info on #260

This PR focuses on fixing the misc and async tests

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/269?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#269** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#268** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#267** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#266** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/266?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#265** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/265?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#264** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/264?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#263** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/263?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#262** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/262?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#260** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/260?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pallets/funding/src/tests/misc.rs [hunk 1]
```diff
@@ -191,6 +214,8 @@ mod helper_functions {
 		let returned_plmc_balances = returned_plmc_mappings.into_iter().map(|map| map.plmc_amount).collect_vec();
 
 		for (expected, calculated) in zip(expected_returns, returned_plmc_balances) {
+			let expected = FixedU128::from_float(expected);
+			let expected = expected.checked_mul_int(PLMC).unwrap();
 			assert_close_enough!(expected, calculated, Perquintill::from_float(0.99));
```
Comment (lrazovic) [inline pallets/funding/src/tests/misc.rs:219]:
```suggestion
for (expected_return, returned_balance) in zip(expected_returns, returned_plmc_balances) {
    let expected_value = FixedU128::from_float(expected_return)
        .checked_mul_int(PLMC)
        .unwrap();

    assert_close_enough!(expected_value, returned_balance, Perquintill::from_float(0.99));
}
```

nit: can we avoid shadowing of variables?

File: pallets/funding/src/tests/misc.rs [hunk 2]
```diff
@@ -245,15 +271,39 @@ mod helper_functions {
...
+			.into_iter()
+			.sorted_by(|a, b| a.account.cmp(&b.account))
+			.map(|map| map.plmc_amount)
+			.collect_vec();
+		let expected_plmc_spent = expected_plmc_spent
+			.into_iter()
+			.sorted_by(|a, b| a.0.cmp(&b.0))
+			.map(|map| {
+				let f64_amount = map.1;
+				let fixed_amount = FixedU128::from_float(f64_amount);
```
Comment (vstam1) [inline pallets/funding/src/tests/misc.rs:300]:
```suggestion
				let fixed_amount = FixedU128::from_float(map.1);
```

**Review by vstam1 — APPROVED** [review]

**Review by lrazovic — APPROVED** [review]

**Comment by JuaniRios** [comment]
### Merge activity

* **Apr 30, 9:52 AM EDT**: @JuaniRios started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Polimec/polimec-node/269).

**Pull request merged**